cron/http.go

```diff
@@ -0,0 +1,338 @@
+package cron
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	apiutil "github.com/datatogether/api/apiutil"
+	flatbuffers "github.com/google/flatbuffers/go"
+	cronfb "github.com/qri-io/qri/cron/cron_fbs"
+)
+
+// HTTPClient implements the Scheduler interface over HTTP, talking to a
+// Cron HTTPServer
+type HTTPClient struct {
+	Addr string
+}
+
+// assert HTTPClient is a Scheduler at compile time
+var _ Scheduler = (*HTTPClient)(nil)
+
+// ErrUnreachable defines errors where the server cannot be reached
+// TODO (b5): consider moving this to qfs
+var ErrUnreachable = fmt.Errorf("cannot establish a connection to the server")
+
+// Ping confirms client can dial the server, if a connection cannot be
+// established at all, Ping will return ErrUnreachable, all other errors
+// will
+func (c HTTPClient) Ping() error {
+	res, err := http.Get(fmt.Sprintf("http://%s", c.Addr))
+	if err != nil {
+		msg := strings.ToLower(err.Error())
+
+		// TODO (b5): a number of errors constitute a service being "unreachable",
+		// we should make a more exhaustive assessment. common errors already covered:
+		// "connect: Connection refused"
+		// "dial tcp: lookup [url] no such host"
+		if strings.Contains(msg, "refused") || strings.Contains(msg, "no such host") {
+			return ErrUnreachable
+		}
+		return err
+	}
+
+	if res.StatusCode == http.StatusOK {
+		return nil
+	}
+	return maybeErrorResponse(res)
+}
+
+// Jobs lists jobs by querying an HTTP server
+func (c HTTPClient) Jobs(ctx context.Context, offset, limit int) ([]*Job, error) {
```

Comment:
hrm, ok looking at other parts of the codebase we have lots of places where methods that respond with a slice structs starts with `List`. I think we should go ahead and make that a new convention:

** **

### Qri List Convention

functions and methods that respond with a slice of structs should start with `List`  followed by the struct name in plural form. eg: `ListJobs`, or `ListDatasets`.

This rule also applies in situations where the receiver of a method uses the plural struct name. As an example a struct called `Jobs` with a method that returns a slice of `Job` should still be called `Jobs.ListJobs()`, not `Jobs.List`. This is for both consistency and to avoid collisions when defining interfaces. Multiple methods named `List` with different signatures negates the possibility of combining two structs in the same interface.

If the List method accepts `offset` and `limit` parameters, the method should be in the form `offset, limit int`, in that order. Passing an `offset` of `0` and a `limit` of `-1` must return the entire list.

** **

How's that sound?

Reply:
Yeah, I like this a lot. Seems like a convention that has come up a lot of places for us, would be good to have it be consistent.

Reply:
Dope, we should probably start assembling our code conventions / style guide somewhere. Tracking Issue: https://github.com/qri-io/qri/issues/761